### PR TITLE
Conform more closely to OGC Processes API spec

### PR DESCRIPTION
**Pull request by drewm-swe**
## Purpose

- This PR revamps the OGC API Processes API implementation to conform more closely to the specification of `OGC API Processes - Part 2: Deploy, Replace, Undeploy`. Future work remains to for the implementation to fully conform to the spec. In its current form, the implementation now provides the ability to request execution of an OGC Process with arbitrary parameters that get pass through to the EMS/ADES (Airflow).
 
## Proposed Changes

- [CHANGE] Entire OGC API Processes implementation to leverage the latest schema provided here: https://github.com/drewm-jpl/ogcapi-processes/tree/master/openapi

## Issues

- https://github.com/unity-sds/unity-sps/issues/129

## Testing

- SPS was successfully deployed with this version of the OGC API server. Following deployment, the `cwl_dag` process was deployed using the OGC API and executed multiple times with different inputs. The different inputs were reflected in the DAG run that was triggered by the execution. Job status was polled using the job status endpoint. Executions/jobs were also dismissed and dismissed jobs were verified to have been deleted from the list of OGC jobs as well as from the list of DAG runs. The `cwl_dag` was also successfully undeployed.

### CODE REVIEW

**Comment by drewm-swe** [comment]
@mike-gangl @LucaCinquini I realize there is a ton of autogenerated cruft in this PR. I want to give you guys a place to focus on, it is the area where endpoints are actually implemented and the methods override an autogenerated inherited class.

These are the two most critical files in the whole PR, they implement the endpoints for jobs and processes:
- https://github.com/unity-sds/unity-sps-ogc-processes-api/pull/15/files#diff-a94e54d1fc85ecb78f6ffca0845e425a4a7693b98ad4f38f34b1a7147fbcf567R1-R217
- https://github.com/unity-sds/unity-sps-ogc-processes-api/pull/15/files#diff-0a8f51412ace8f9d1f882d885e05d597e25723416af722cad18339c720ddd279R1-R209

File: app/src/openapi_server/impl/processes_api.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+                crud.create_job(self.db, job.model_dump(by_alias=True))
+                if prefer == "respond-async":
+                    # Asynchronous execution
+                    return job
+                else:
+                    # Synchronous execution
+                    # Note: In a real-world scenario, you'd wait for the job to complete
+                    # and return the actual results. This is a simplified version.
+                    # TODO get result from job using the result endpoint and return it here.
+                    return Execute200Response(
```
Comment (mike-gangl) [inline app/src/openapi_server/impl/processes_api.py:195]: we're assuming we never call the sync mode here, correct? I'm fine with that- just need to be clear in documentation/conformance.

File: app/src/openapi_server/impl/jobs_api.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+                links=job.links,
+            )
+            for job in jobs
+        ]
+        return JobList(
+            jobs=job_status_infos,
+            links=[],
+        )
+
+    def get_result(self, jobId: str, prefer: str) -> Dict[str, InlineOrRefData]:
```
Comment (mike-gangl) [inline app/src/openapi_server/impl/jobs_api.py:124]: i'm curious what this means in the airflow/cwl-dag context. With CWL, we can define the output of a workflow or CommandLinetool, and can have a lot of control over that w.r.t. application-packages and what _they_ should return. 

Thinking ahead, if we run stage_out, i wonder if it makes sense to publish the 'success_features.json' as the output for cwl_dag. This is future work, not something to address here, jsut more architecturally.

**Comment by mike-gangl** [comment]
So you're right, this is a ton to cover and understand from an "approval" or "reject" stand point. I'll approve this here, but we should be aware that we might need to come back and make updates to this code base once we integrate with unity-sds-client and the system-test.

The next steps should be: add a system test to the [wpst.features](https://github.com/unity-sds/unity-monorepo/blob/main/tests/system-tests/features/sps/wpst.feature) (probably rename that) and the corresponding [steps](https://github.com/unity-sds/unity-monorepo/blob/main/tests/system-tests/features/steps/sps/steps.py). which utilizes the unity-sds-client to make the OGC calls to deploy, execute, status, dismiss, etc a job. that would validate that the integration is working in a known state with (multiple) application packages being sent through cwl_dag.

**Review by mike-gangl — APPROVED** [review]
i have some comments, but without the end to end validation tests (i think that's what the OGC Notebook is doing) it's hard to review this in a vaccum.

**Pull request merged**